crates/shim/src/asynchronous/publisher.rs

```diff
@@ -18,34 +18,110 @@ use std::os::unix::io::RawFd;
 
 use async_trait::async_trait;
 use containerd_shim_protos::{
-    api::Empty,
+    api::{Empty, Envelope},
     protobuf::MessageDyn,
     shim::events,
     shim_async::{Client, Events, EventsClient},
     ttrpc,
     ttrpc::{context::Context, r#async::TtrpcContext},
 };
+use log::debug;
+use tokio::sync::{mpsc, Mutex};
 
 use crate::{
-    error::Result,
+    asynchronous::Arc,
+    error::{self, Result},
     util::{asyncify, connect, convert_to_any, timestamp},
 };
+//Mas queue size
+const QUEUE_SIZE: i64 = 1024;
```

Comment:
Please use rust's doc here.

Reply:
What I meant here is that comment should start with `///` or `//!` to be properly recognized by rust's doc system.

The one that you added (`#doc = "..."`) should probably just go to the file's top doc comment (`//!`) to represent module documentation.